compute/client_library/ingredients/instances/change_machine_type.py

```diff
@@ -0,0 +1,54 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# This is an ingredient file. It is not meant to be run directly. Check the samples/snippets
+# folder for complete code samples that are ready to be used.
+# Disabling flake8 for the ingredients file, as it would fail F821 - undefined name check.
+# flake8: noqa
+
+from typing import NoReturn
+
+from google.cloud import compute_v1
+
+
+# <INGREDIENT change_machine_type>
+def change_machine_type(project_id: str, zone: str, instance_name: str, new_machine_type: str) -> NoReturn:
+    """
+    Changes the machine type of compute instance. The instance needs to be in `TERMINATED` state for this
```

Comment:
Please consider using 'VM' when referring to instances. We're changing this in our docs too.

Changes the machine type of a VM. The VM needs to be in the 'TERMINATED' state for this operation to be successful.